lib/apiservers/engine/backends/container/viccontainer.go

```diff
@@ -65,3 +70,24 @@ func (v *VicContainer) List() []string {
 	v.m.RUnlock()
 	return IDs
 }
+
+// Tries to lock the container.  Timeout argument defines how long the lock
+// attempt will be tried.  Returns true if locked, false if timed out.
+func (v *VicContainer) TryLock(timeout time.Duration) bool {
+	timeChan := time.After(timeout)
+	select {
+	case <-timeChan:
+		return false
+	case v.lockChan<-true:
+		return true
+	}
+}
+
+// Unlocks the container
+func (v *VicContainer) Unlock() {
+	select {
+	case <-v.lockChan:
+	default:
+		panic()
```

Comment:
Could you panic with comment? like panic("Attempt to unlock VicContainer more than once.")